lib/airbrake/sneakers.rb

```diff
@@ -0,0 +1,18 @@
+module Airbrake
+  module Sneakers
+    ##
+    # Provides integration with Sneakers.
+    #
+    # @see https://github.com/jondot/sneakers
+    class ErrorReporter
+      def call(exception, worker = nil, **context)
```

Comment:
Is double asterisk intentional?

Reply:
Isn't it for keyword arguments? Why can't we use a single asterisk?

Reply:
I simply want to support the call without a given worker as second argument:
```ruby
opts = { foo: 'bar' }
ErrorReporter.call(exception, opts)
```
so now with double asterisk `worker` is `nil` and `context` is `{ foo: 'bar' }`.
If I had just used a single asterisk on `context` the options would get assigned to `worker` and `context` is `nil`.

They added the `worker` parameter and I want to provide a solution that works with and without a given `worker` as second parameter.

This only works if the given options hash has symbolized keys, but this is the case:
https://github.com/jondot/sneakers/blob/7c7697757af239e2232e34c31bfd48e8bdacd596/lib/sneakers/worker.rb#L72

Another way would be to check the type of the worker argument, but I think it is more ugly.
```ruby
def call(exception, worker, context = nil)
  if worker.is_a?(Hash)
    context = worker
    worker = nil
  end
  # ...
end
```

Reply:
That makes sense. Can you please provide an example (just a link to the code) of a real job with keyword arguments?

Reply:
By the way, I am fairly confident these details must be documented as a test. It's very tricky to understand so we must be sure that everything works as expected. Could you please a test(s) to make sure we support multiple arguments (different combinations).

Reply:
The question is: Can or should we test multiple sneakers versions? It is easy to use the unreleased git version locally by adding `gem 'sneakers', github: 'jondot/sneakers'` to the Gemfile, but how could we manage that in a test?

I think I don't know, what you mean by "real job with keyword arguments". The keyword arguments are constructed here:
https://github.com/jondot/sneakers/blob/7c7697757af239e2232e34c31bfd48e8bdacd596/lib/sneakers/worker.rb#L72
and here the error_reporter is just called with it here:
https://github.com/jondot/sneakers/blob/7c7697757af239e2232e34c31bfd48e8bdacd596/lib/sneakers/error_reporter.rb#L18

But yeah, I can add a test, where the `Airbrake::Sneakers::ErrorReporter` is called directly with different arguments. Would that help?

Reply:
Testing this project is already hard enough. For simplicity we test only latest stable versions of libraries. So to answer your question, no we shouldn't.

Ok, I figured it out, let's stick to `**`.